### PR TITLE
Responsive Meta Panel

### LINKED ISSUE #53: Fixed sidebar
Make the sidebar of information pages containing metadata and table of contents (TOC) fix, so it is always reachable.

Notice:

- The menu may get larger
- Make it responsive
- The sidebar can get longer than the height of the screen
- Maybe it is nice to have a collapsable TOC?
- Ensure proper accessibility of TOC, for screenreaders and via keyboard

### LINKED ISSUE #275: Proper layout for minutes on mobile
Minutes use a two-column layout for content and meta data. On mobile, this is not really usable.
Instead, the meta data on mobile should be in a collabsable drawer.

### LINKED ISSUE #329: Navigating using TOC does not account for navbar
When navigating to a part of the page via the toc on the side, the headline of that content is at the top of the page. However, it can not be seen as it is hidden behind the navbar.

The navigation should bring the view to the part of the page so that the headline is visibile, directly beneath the navbar.

### PR DESCRIPTION
**Pull request by SilvanVerhoeven**
Closes #275, closes #329, closes #53
Replaces stale #285

### CODE REVIEW

File: myhpi/core/templates/core/minutes.html [hunk 1]
```diff
@@ -6,82 +6,85 @@
...
+        </aside>
+        {% include "core/sidebar.html" %}
+    </div>
+    <div id="minutes-footer" class="d-none d-print-block"></div>
+    <div id="minutes-navigation" class="row d-print-none">
+        <div class="col-4">
+            {% prev_minutes page as prev %}
+            {% if prev %}
+            <a href="{{ prev.url }}" class="btn btn-outline-primary minutes-navigation-button">{% bs_icon 'arrow-left'
+                extra_classes='minutes-navigation-icon' %} {% translate "Previous minutes" %}</a>
```
Comment (frcroth) [inline myhpi/core/templates/core/minutes.html:74]: Looks like this for me:
image

File: myhpi/static/js/myHPI.js [hunk 2]
```diff
@@ -38,75 +38,80 @@ const toggleHideOnScrollBlock = (element) => {
...
+      day: "numeric",
+      hour: "numeric",
+      minute: "2-digit",
+      timeZoneName: "short",
+    })
+
+    // replace <time> width <abbr> tag
+    let abbr = document.createElement("abbr")
+    abbr.setAttribute("title", lastPublishedLocalized)
+    lastPublished.removeAttribute("title")
```
Comment (frcroth) [inline myhpi/static/js/myHPI.js:77]: What does that do?

File: myhpi/static/js/navbar.js [hunk 3]
```diff
@@ -17,7 +17,7 @@
  * @returns True if it is collapsed, false otherwise.
```
Comment (frcroth) [inline myhpi/static/js/navbar.js:17]: If you used some specifc linter (settings), can you include them in the README?

**Review by frcroth — COMMENTED** [review]
Headings on minutes look like this for me:
image

File: myhpi/core/templates/core/minutes.html [hunk 4]
```diff
@@ -6,82 +6,85 @@
...
-        {% endwith %}
     </div>
+    {% endwith %}
+</div>
 {% endblock %}
 
 {% block extra_js %}
 <script type="text/javascript" src="{% static 'js/print_processor.js' %}"></script>
-{% endblock %}
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/minutes.html:90]: That's what my formatter does. I am very open for suggestions for a formatter/formatter config that does not f*** up the whole formatting

File: myhpi/static/js/navbar.js (hunk 3 above)
Comment (SilvanVerhoeven) [reply myhpi/static/js/navbar.js:17]: For JS I can, but for CSS/HTML I'd rather get those in return ^^

**Comment by SilvanVerhoeven** [comment]
> Headings on minutes look like this for me: image

Correct, those are permalinks to the headings (so one can share a link that jumps to specific section on the page)

File: myhpi/core/templates/core/minutes.html (hunk 1 above)
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/minutes.html:74]: Yeah I tweaked the design a bit /s

File: myhpi/static/js/myHPI.js (hunk 2 above)
Comment (SilvanVerhoeven) [reply myhpi/static/js/myHPI.js:77]: Yeah so I didn't do anything in this file, except for my formatter to change the format

Comment (frcroth) [reply myhpi/static/js/myHPI.js:77]: This line seems to be missing:` lastPublished.innerText = lastPublished.textContent.trim()`

**Comment by jeriox** [comment]
grafik
the content is not using the full width of the screen here

**Comment by jeriox** [comment]
> Correct, those are permalinks to the headings (so one can share a link that jumps to specific section on the page)

I like the idea of having those, but I'm also not so happy with the proposed layout. I don't think that most people get that the # _after_ the title is meant to be the permalink. And the arrow-out-of-the-box-icon is only used for external links, but not internal ones. So I think its odd that is now part of another internal link.

File: myhpi/core/templates/core/minutes.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (jeriox) [reply myhpi/core/templates/core/minutes.html:90]: over at ephios, we use https://github.com/rtts/djhtml

[the remark this answers is not in the document]
Comment (frcroth) [reply myhpi/core/templates/core/minutes.html:90]: Yes, I think it would be nice to include that

**Comment by abc013** [comment]
> I like the idea of having those, but I'm also not so happy with the proposed layout. I don't think that most people get that the # after the title is meant to be the permalink. And the arrow-out-of-the-box-icon is only used for external links, but not internal ones. So I think its odd that is now part of another internal link.

(Just to give some additional Senf to the Wurst)

Personally, I really like GitHub's approach to that: Just showing a 'link' icon behind/before the title should be enough imho (It's hover-on on GitHub, but doesn't have to be here).
image

**Comment by SilvanVerhoeven** [comment]
> the arrow-out-of-the-box-icon is only used for external links, but not internal ones. So I think its odd that is now part of another internal link.

@jeriox Agreed, but I am not sure I can do much about this at that point. Currently, all links not created by internal references get this arrow icon. I think this should be reworked in general in another branch

File: myhpi/core/templates/core/minutes.html (hunk 1 above)
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/minutes.html:74]: Fixed

File: myhpi/core/templates/core/minutes.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/minutes.html:90]: Moved into a new issue

File: myhpi/static/js/myHPI.js (hunk 2 above)
Comment (SilvanVerhoeven) [reply myhpi/static/js/myHPI.js:77]: Thank you all for your help to fix this, done :)

File: myhpi/static/js/navbar.js (hunk 3 above)
Comment (SilvanVerhoeven) [reply myhpi/static/js/navbar.js:17]: Added prettier into the pre-commit hook

**Comment by SilvanVerhoeven** [comment]
> Personally, I really like GitHub's approach to that: Just showing a 'link' icon behind/before the title should be enough imho (It's hover-on on GitHub, but doesn't have to be here).

Unfortunately, the used permanent link extension only allows to insert strings. The UTF 8 link emoji looks horrible on the latest windows 10, that's why we keep the `#` for now.

File: myhpi/core/templates/core/sidebar.html [hunk 5]
```diff
@@ -3,37 +3,63 @@
...
         {% endif %}
     {% endwith %}
 {% endif %}
-{% with parsed_md.1 as toc %}
-    {% if toc|non_empty %}
-        <aside class="side-panel border-accent toc-container d-print-none">
-            <h1 class="side-panel-title">{% translate "Table of contents" %}</h1>
-            {{ toc }}
+
+<!-- Bootstrap 5.2 supports responsive offcanvas.
```
Comment (jeriox) [inline myhpi/core/templates/core/sidebar.html:20]: @dasGoogle

**Review by abc013 — CHANGES_REQUESTED** [review]
There seem to be some issues, nonetheless :(
image
Also looks like the icons are cut off at the edges
image

**Comment by abc013** [comment]
Poll pages still seem to have the old layout in the smartphone version

**Review by abc013 — COMMENTED** [review]
otherwise looking good!

**Review by jeriox — APPROVED** [review]

**Comment by SilvanVerhoeven** [comment]
> Abbreveations overlay button

Opened #587 for this

> Look like the icons are cut off at the edges

This varies depending on screen size, resolution and browser. This is probably not an issue introduced by this PR

**Comment by SilvanVerhoeven** [comment]
> Poll pages still seem to have the old layout in the smartphone version

Opened #579 for this

**Pull request merged**